### PR TITLE
Expand configuration docs (fixes GH-2560)

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
We could expand on this more showing the ``sentry config`` usage basics but this is a big step forward

File: docs/config.rst [hunk 1]
```diff
@@ -9,16 +9,79 @@ First Install
...
+
+    ::
+
+        system.secret-key: 'a-really-long-secret-value'
+
+    To generate a new value, you can use Python:
+
+    .. code-block:: python
+
+        # $ sentry shell
```
Comment (mattrobenolt) [inline docs/config.rst]: In the docker container, I added a helper `generate-secret-key`. Should we elevate this inside the actual `sentry` cli?

Comment (mattrobenolt) [inline docs/config.rst]: See: https://github.com/getsentry/docker-sentry/commit/e0c879f331da7bd0ddc93e7033297548a8f88a00

Comment (dcramer) [inline docs/config.rst]: Yeah I was thinking about that too. Maybe ``config generate-secret-key`` and it just outputs to stdout? Could also have a prompt to fill it in for the existing file.

**Comment by mattrobenolt** [comment]
This looks good. :+1:

File: docs/config.rst (hunk 1 above)
Comment (mattrobenolt) [inline docs/config.rst]: I would just do print to stdout. But I can add `config generate-secret-key`. That seems reasonable.

**Pull request merged**